src/commands/hook/hook.js

```diff
@@ -4,7 +4,7 @@ const HOOK: Object = {
   FILENAME: 'prepare-commit-msg',
   CONTENTS:
     '#!/bin/sh\n# gitmoji as a commit hook\n' +
-    'exec < /dev/tty\ngitmoji --hook $1 $2\n'
+    'exec < /dev/tty\nnpx gitmoji --hook $1 $2\n'
```

Comment:
Hey!

Let's change this to:

```suggestion
    'exec < /dev/tty\nnpx -c "gitmoji" --hook $1 $2\n'
```

The package name published on the registry is `gitmoji-cli` therefore if we use `npx gitmoji` the command will suggest installing `gitmoji` which is a different package 😅 

Making this change will allow the hook resolving the `gitmoji` cli command either as a local dependency or a global one

Reply:
> ah right.. if it is not installed locally or globally it won't work. But I think it should be
> 
> ```shell
> npx --package=gitmoji-cli gitmoji
> ```

further update for the script:

```sh
#!/usr/bin/env sh                                                                                                                                                                                             

if npx -v >/dev/null
then  npx --package=gitmoji-cli gitmoji --hook $1 $2
else  gitmoji --hook $1 $2
fi

```

- small change in shebang to make it more portable
- check in the hook script
  - is `npx` installed -> use `npx` 
  - not installed -> use global installed version

Reply:
@carloscuesta updated the code.
Now using `npx -c gitmoji` if npx is installed. Else fallback to `gitmoji`.
But another thought. Think it makes sense to change the check to `if npx -c gitmoji -v >/dev/null `. Then it checks in addiition if `gitmoji` can be called with `npx -c`.
Currently am not sure if there is an issue if e.g. `npx` is installed and `gitmoji` is installed with e.g. `brew`.